Add release notes for 0.238 (#14770)
prestodb/presto: presto-docs/src/main/sphinx/release/release-0.238.rst

=============
Release 0.238
=============

General Changes
_______________
* Fix case sensitivity issue in SQL function parameters. Function parameters should be treated case-insensitive.
* Add ``target_result_size`` session property to customize data batch sizes being streamed from coordinator.

> * Add optimization to push null filters to the INNER side of equijoins. The optimization can be enabled with `optimize-nulls-in-joins`.

Reviewer: Use pairs of double back quotes instead of single back quotes for proper formatting. Here and below. https://github.com/prestodb/presto/wiki/Release-Notes-Guidelines#formatting

* Add sessionProperty override to Presto JDBC URI.

> * Add support to create external function (this does not include external function execution).

Reviewer: Remove the content in the parenthesis. We introduce what is included in the release, not what is not included.

> * Remove max buffer count config property `driver.max-page-partitioning-buffer-count` for optimized repartitioning.

Reviewer: s / config property / configuration property for consistency

Verifier Changes
________________
* Fix an issue where session properties of control and test queries also affects checksum queries.
* Add support to report peak task memory usage for control and test queries.

Geospatial Changes
__________________

> * Fix NPE in :func:`geometry_invalid_reason`.

Reviewer 2: NPE is Java-specific.  Let's say "fix error in geometry_invalid_reason"

* Fix integer overflow in certain cases with Bing Tiles.

Hive Changes
____________
* Fix a bug in Parquet reader which manifests when there are nested column schema changes.
* Add support for exchange materialization of table bucketed by non-hive types. This can be enabled using value ``PRESTO_NATIVE`` for the ``bucket_function_type_for_exchange`` session property  or the ``hive.bucket-function-type-for-exchange `` configuration property.

Pinot Changes
_____________

> * Adding config: `pinot.mark-data-fetch-exceptions-as-retriable` to let presto retry on pinot data fetcher related exceptions.

Reviewer: Use the suggested phrasing https://github.com/prestodb/presto/wiki/Release-Notes-Guidelines#phrasing
```
Add support to retry data fetch exception. This can be enabled by setting the configuration
property ``pinot.mark-data-fetch-exceptions-as-retriable``.
```

> * Support predicate pushdown for literals of type `DATE`, `TIMESTAMP`, or `TIMESTAMP_WITH_TIME_ZONE`.

Reviewer: ```
Add predicate pushdown support for ``DATE``, ``TIMESTAMP``, and ``TIMESTAMP_WITH_TIME_ZONE``
literals.
```

=============
Release 0.238
=============

**Highlights**
==============

> * Fix case sensitivity issue in SQL function parameters. Function parameters should be treated case-insensitive.

Reviewer 2: ```suggestion
* Fix SQL function parameters to be case-insensitive.
```

> * Add support to create external function.

Reviewer 3: Just external function is not. Lets add a reference to a Github issue to explain what it does. @rongrong

Author: linking issue.

> * Add predicate pushdown support for ``DATE``, ``TIMESTAMP``, and ``TIMESTAMP_WITH_TIME_ZONE`` literals.

Reviewer 3: This is only for the Pinot connector but this is not clear from the highlights.

General Changes
_______________
* Fix case sensitivity issue in SQL function parameters. Function parameters should be treated case-insensitive.
* Add ``target_result_size`` session property to customize data batch sizes being streamed from coordinator.
* Add optimization to push null filters to the INNER side of equijoins. The optimization can be enabled with ``optimize-nulls-in-joins``.

> * Add sessionProperty override to Presto JDBC URI.

Reviewer 3: If session property is intended to be an identifier then we should use double back quotes or just say `session property`

* Add support to create external function.
* Remove max buffer count configuration property ``driver.max-page-partitioning-buffer-count`` for optimized repartitioning.

> Verifier Changes

Reviewer 3: Connectors come before verifier - https://github.com/prestodb/presto/wiki/Release-Notes-Guidelines

________________
* Fix an issue where session properties of control and test queries also affects checksum queries.
* Add support to report peak task memory usage for control and test queries.

Geospatial Changes
__________________
* Fix error in geometry_invalid_reason.
* Fix integer overflow in certain cases with Bing Tiles.

Hive Changes
____________
* Fix a bug in Parquet reader which manifests when there are nested column schema changes.

> * Add support for exchange materialization of table bucketed by non-hive types. This can be enabled using value ``PRESTO_NATIVE`` for the ``bucket_function_type_for_exchange`` session property  or the ``hive.bucket-function-type-for-exchange `` configuration property.

Reviewer 2: I would make this a highlight?

> * Add support for exchange materialization of table bucketed by non-hive types. This can be enabled using value ``PRESTO_NATIVE`` for the ``bucket_function_type_for_exchange`` session property  or the ``hive.bucket-function-type-for-exchange `` configuration property.

Reviewer 3: Following the convention in https://github.com/prestodb/presto/wiki/Release-Notes-Guidelines#phrasing - I will rework the second line as `This can be enabled by setting the abc session property or xyz configuration property to mno`

=============
Release 0.238
=============

**Highlights**
==============
* Fix SQL function parameters to be case-insensitive.
* Add support to create external function. (:issue:`13254`)
* Add support for exchange materialization of table bucketed by non-hive types. This can be enabled by setting the ``bucket_function_type_for_exchange`` session property or ``hive.bucket-function-type-for-exchange `` configuration property to ``PRESTO_NATIVE``

> * Add predicate pushdown support for ``DATE``, ``TIMESTAMP``, and ``TIMESTAMP_WITH_TIME_ZONE`` literals for Pinot.

Reviewer 3: Pinot connector

General Changes
_______________

> * Fix case sensitivity issue in SQL function parameters. Function parameters should be treated case-insensitive.

Reviewer 3: I prefer the highlight version of this - `Fix SQL function parameters to be case-insensitive.`

* Add ``target_result_size`` session property to customize data batch sizes being streamed from coordinator.
* Add optimization to push null filters to the INNER side of equijoins. The optimization can be enabled with ``optimize-nulls-in-joins``.
* Add ``sessionProperty`` override to Presto JDBC URI.
* Add support to create external function. (:issue:`13254`)
* Remove max buffer count configuration property ``driver.max-page-partitioning-buffer-count`` for optimized repartitioning.

Geospatial Changes
__________________

> * Fix error in :func:geometry_invalid_reason.

Reviewer 3: So github nicely formatted my comment - please see https://github.com/prestodb/presto/wiki/Release-Notes-Guidelines#formatting for how to format function names

> * Fix error in geometry_invalid_reason.

Reviewer 3: :func:`geometry_invalid_reason`

=============
Release 0.238
=============

**Highlights**
==============
* Fix SQL function parameters to be case-insensitive.
* Add support to create external function. (:issue:`13254`)
* Add support for exchange materialization of table bucketed by non-hive types. This can be enabled by setting the ``bucket_function_type_for_exchange`` session property or ``hive.bucket-function-type-for-exchange `` configuration property to ``PRESTO_NATIVE``
* Add predicate pushdown support for ``DATE``, ``TIMESTAMP``, and ``TIMESTAMP_WITH_TIME_ZONE`` literals for Pinot connector.

General Changes
_______________
* Fix SQL function parameters to be case-insensitive.
* Add ``target_result_size`` session property to customize data batch sizes being streamed from coordinator.
* Add optimization to push null filters to the INNER side of equijoins. The optimization can be enabled with ``optimize-nulls-in-joins``.
* Add ``sessionProperty`` override to Presto JDBC URI.
* Add support to create external function. (:issue:`13254`)
* Remove max buffer count configuration property ``driver.max-page-partitioning-buffer-count`` for optimized repartitioning.

> * A new session property ``query_max_broadcast_memory`` that allows for specifying the maximum amount of memory a query can use for broadcast join.

Reviewer: I would say
```
Add session property ``query_max_broadcast_memory`` to limit the memory a query can use for broadcast join.
```

Geospatial Changes
__________________
* Fix error in :func:`geometry_invalid_reason`.
* Fix integer overflow in certain cases with Bing Tiles.

Hive Changes
____________
* Fix a bug in Parquet reader which manifests when there are nested column schema changes.

> * Add support for exchange materialization of table bucketed by non-hive types. This can be enabled by setting the ``bucket_function_type_for_exchange`` session property or ``hive.bucket-function-type-for-exchange `` configuration property to ``PRESTO_NATIVE``

Reviewer: missing period at the end of the sentence (".")

Pinot Changes
_____________
* Add support to retry data fetch exception. This can be enabled by setting the configuration property ``pinot.mark-data-fetch-exceptions-as-retriable``.
* Add predicate pushdown support for ``DATE``, ``TIMESTAMP``, and ``TIMESTAMP_WITH_TIME_ZONE`` literals.

Verifier Changes
________________
* Fix an issue where session properties of control and test queries also affects checksum queries.
* Add support to report peak task memory usage for control and test queries.

> 

Reviewer: remove extra empty line
